sigmf/sigmffile.py

```diff
@@ -890,14 +944,23 @@ def get_dataset_filename_from_metadata(meta_fn, metadata=None):
     return None
 
 
-def fromarchive(archive_path, dir=None):
-    """Extract an archive and return a SigMFFile.
+def fromarchive(archive_path):
+    """Extract an archive and return containing SigMFFiles.
 
-    The `dir` parameter is no longer used as this function has been changed to
-    access SigMF archives without extracting them.
+    If the archive contains a single recording, a single SigMFFile object will
```

Comment:
This seems "squishy" to me -- I don't think I'm very comfortable with the prospect of the return type changing based on the contents of the SigMF Archive.
However I do acknowledge that it is an attempt to provide some level of backward compatibility with existing code.